mmdet/models/dense_heads/base_dense_head.py

```diff
@@ -14,10 +18,264 @@ def loss(self, **kwargs):
+                    max_scores, _ = (scores_ *
+                                     score_factor[:, None]).max(dim=1)
+                else:
+                    max_scores, _ = scores_.max(dim=1)
+                _, topk_inds = max_scores.topk(nms_pre)
+
+                priors = self.prior_generator.sparse_priors(
+                    topk_inds, featmap_size_hw, level_idx, scores.dtype,
+                    scores.device)
+                bbox_pred = bbox_pred[topk_inds, :]
+                scores = scores[topk_inds, :]
+                if with_score_factors:
+                    score_factor = score_factor[topk_inds]
+            else:
+                priors = self.prior_generator.single_level_grid_priors(
+                    featmap_size_hw, level_idx, scores.device)
+
+            bboxes = self.bbox_coder.decode(
+                priors, bbox_pred, max_shape=img_shape)
+            mlvl_bboxes.append(bboxes)
+            mlvl_scores.append(scores)
+            if with_score_factors:
+                mlvl_score_factor.append(score_factor)
+
+        return self._bbox_post_process(mlvl_scores, mlvl_bboxes,
+                                       img_meta['scale_factor'], cfg, rescale,
+                                       with_nms, mlvl_score_factor, **kwargs)
+
+    def _bbox_post_process(self,
+                           mlvl_scores,
+                           mlvl_bboxes,
+                           scale_factor,
+                           cfg,
+                           rescale=False,
+                           with_nms=True,
+                           mlvl_score_factor=None,
+                           **kwargs):
+        """bbox post-processing method.
+
+        It usually completes the rescale to the original image scale and nms
+        operation. if with_nms is True, Perform nms operation on bbox and
+        return det_bboxes and det_labels. if with_nms is False and
+        mlvl_score_factor is None, return mlvl_bboxes and  mlvl_scores, else
+        return mlvl_bboxes,  mlvl_scores and  mlvl_score_factor. Usually
+        with_nms is False is used for aug test.
+
+        Args:
+            mlvl_scores (list[Tensor]): Box scores for a single image
+                Has shape (num, num_class).
+            mlvl_bboxes (list[Tensor]): Box energies / deltas for a single
+                image with shape (num, 4).
+            scale_factor (ndarray, optional): Scale factor of the image arange
+                as (w_scale, h_scale, w_scale, h_scale).
+            cfg (mmcv.Config): Test / postprocessing configuration,
+                if None, test_cfg would be used.
+            rescale (bool): If True, return boxes in original image space.
+                Default: False.
+            with_nms (bool): If True, do nms before return boxes.
+                Default: True.
+            mlvl_score_factor (list[Tensor], optional):  score_factor for each
+                image with shape (num, ). Default: None.
+
+        Returns:
```

Comment:
Results of detected boxes and labels. ``with_nms`` would always be ``False`` except for doing the AugTest.  
It return ``mlvl_bboxes`` and ``mlvl_scores`` when ``with_nms`` is True, otherwise the `mlvl_score_factor` would be added.